### ISSUE TITLE
Deprecate acceptsKeyboardFocus

### ISSUE THREAD

**Issue by chrisglein** [microsoft/react-native-macos#498]
From [Keyboarding in React Native Desktop](https://github.com/microsoft/react-native-windows/pull/5590/files) document:

> React Native core introduced focusable that does the same thing as acceptsKeyboardFocus in v0.63. Windows has added focusable and marked acceptsKeyboardFocus as deprecated in 0.63. macOS needs to do the same.

**Comment by NickGerleman** [comment]
Updating the original description, since the RNW deprecation ended up happening in 0.63. The version matrix basically looked like:

- RNW 0.63 added focusable and runtime warnings if acceptsKeyboardFocus is used
- RNW 0.64 will remove acceptsKeyboardFocus entirely (likely going to have a debug assert if VMs see them)
- For Office, we need to support bundles built against N-1 versions of the react-native version we have in repo, so react-native-win32 also removed typings in 0.63

The TLDR: react-native-macos will need to add focusable by 0.63, and will be able to fully remove acceptsKeyboardFocus in 0.64. Deprecation warnings at runtime are useful since folks are using acceptsKeyboardFocus without typings.
